pymedphys/_experimental/pseudonymisation/__init__.py

```diff
@@ -134,3 +138,67 @@ def is_valid_strategy_for_keywords(
         if vr not in replacement_strategy:
             return False
     return True
+
+
+def pseudonymise(dicom_input, output_path=None):
+    """Convenient API to pseudonymisation.
+    Elements whose tags are not in the pydicom dictionary will be deleted
+    PatientSex will not be modified/pseudonymised
+    For fine tune control, use anonymise_dataset() instead
+
+    Parameters
+    ----------
+    dicom_input : ``pydicom.dataset.Dataset | str | pathlib.Path``
+        Either a dataset, a path to a file or a path to a directory
+    output_path : ``str | pathlib.Path``, optional
+        If the input is a file or a path, the directory to place the
+        pseudonymised files, by default None
+
+    Returns
+    -------
+    ``pydicom.dataset.Dataset`` | ``str`` | ``list`` of ``str``
+        if the dicom_input was a dataset, return the pseudonymised dataset
+        if the dicom input was a file, return the path to the pseudonymised file.
+        if the dicom input was a directory, return the list of successfully
+        anonymised files, and return that instead of None
+    """
+    replacement_strategy = strategy.pseudonymisation_dispatch
+    identifying_keywords_for_pseudo = get_default_pseudonymisation_keywords()
+    if not is_valid_strategy_for_keywords():
+        logging.error("Pseudonymisation strategy is not valid for keywords")
+        logging.error("Please submit issue to PyMedPhys")
+        # but continue on, the data might not contain the offending keywords
+        # and if it does... there will be some kind of error raised
+    keywords_to_leave_unchanged = list("PatientSex")
+
+    if isinstance(dicom_input, pydicom.dataset.Dataset):
+        pseudo_ds = anonymise_dataset(
+            dicom_input,
+            keywords_to_leave_unchanged=keywords_to_leave_unchanged,
+            delete_unknown_tags=True,
```

Comment:
Given this flag is reused below, it might be worth defining just before this line and then reusing the defined flag.

Reply:
if you are referring to delete_unknown tags...  I don't see the point in using a separate variable that is inherently interned.  If you are referring to something else, I need you to specify it explicitly.  I am re-using the variables that require a function call.

Reply:
I just mean, within the function itself (not as a part of its API) have:

```python
		delete_unknown_tags = True
		
		# ...

        pseudo_ds = anonymise_dataset(
            dicom_input,
            keywords_to_leave_unchanged=keywords_to_leave_unchanged,
            delete_unknown_tags=delete_unknown_tags,  # use defined flag
            replacement_strategy=replacement_strategy,
            identifying_keywords=identifying_keywords_for_pseudo,
        )

        # ...

        pseudonymised_file_list = anonymise_directory(
            dicom_input,
            output_dirpath=output_path,
            keywords_to_leave_unchanged=keywords_to_leave_unchanged,
            delete_unknown_tags=delete_unknown_tags,  # use defined flag
            replacement_strategy=replacement_strategy,
            identifying_keywords=identifying_keywords_for_pseudo,
        )

        # ...

        pseudonymised_filepath = anonymise_file(
            dicom_input,
            output_filepath=output_path,
            keywords_to_leave_unchanged=keywords_to_leave_unchanged,
            delete_unknown_tags=delete_unknown_tags,  # use defined flag,
            replacement_strategy=replacement_strategy,
            identifying_keywords=identifying_keywords_for_pseudo,
        )
```

Reply:
I respectfully disagree.
It's just as easy to type True each time (easier, actually), and it makes the intent clear without having to refer up.
If this were a string literal, I'd agree wholeheartedly.  But a constant boolean that is interned...

Reply:
All good, this was certainly nit-picking anyhoo.